microcosm_flask/decorators/schemas.py

```diff
@@ -41,6 +41,35 @@ def get_associated_schema(schema_cls, name_suffix):
     raise KeyError(f"Schema {schema_cls} does not have an associated schema with suffix {name_suffix}")
 
 
+def set_associated_schema(target_cls, name_suffix, associated_schema):
+    """
+    Add a derived schema as an attribute to a schema class, using conventions
+    that will result in the associated schema being exposed in the OpenAPI definition of the service.
+
+    """
+    # Use the class name in the attribute name to avoid sharing with children classes
+    attr_name = associated_schemas_attr_name(target_cls)
+    try:
+        associated_schemas = getattr(target_cls, attr_name)
+        if name_suffix in associated_schemas:
+            raise ValueError(f"Schema {target_cls} already has an associated schema for suffix {name_suffix}")
+        associated_schemas[name_suffix] = associated_schema
+    except AttributeError:
```

Comment:
Can this be tighter in scope only encompassing `associated_schemas = getattr(target_cls, attr_name)`? As that should be only line we expect an `AttributeError` from.

Reply:
Reproducing IRL conversation here:
- There isn't really a way to tighten the scope while using `try/catch`
- But I think that's acceptable, situations like that are a bit of a side effect of using exceptions for control flow.